### PR TITLE
EmptyStateView component, refactored dashboard layout

### PR DESCRIPTION
**Pull request by kara-o**
## What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
*	Main purpose of this PR was to add the ‘No Donations’ view for both client and donor.  The user story is that client/donor logs in, and once taken to the dashboard ‘home’ screen, if there are no donations available (for client) or no active donations made (for donor), then the screen shows an image with text along the lines of ‘No Donations Available.’
*	This is the view for an ‘empty state’ situation – in this case, the array of donations we received from the db query is empty.  I decided to create a reusable component for this situation.  For any empty state situation, in which there is nothing to show the user, we can use this component which will display the banana logo along with our choice of text:

clientdb
donordb

## Other changes in this PR:
*	Updated background color of DashboardScreen/DonorDashboardScreen to match Figma (will need to decide background of SafeAreaView to match)
*	Removed padding from both dashboard screens as discussed with Tao (global horizontal padding causing issues)
*	Changed where we had Title component, instead of being coupled with Navbar, I separated the two so Navbar is alone, and Title is with the rest of the Dashboard content (in one container).  This made more sense to me based on layout, and I was able to then add padding to just Dashboard content to make it fit with Figma and not affect Header/Navbar.

Screen Shot 2020-05-02 at 12 41 13 PM

*	Added DonorDonationScreen to MainStack navigator for functionality of “+” button on DonorDashboardScreen to work

Screen Shot 2020-05-02 at 12 51 06 PM

## Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
I tested out my code with the new NavBar/SafeAreaView changes and they work well with no breaks.

### CODE REVIEW

**Comment by duskcloudxu** [comment]
I am trying to test the page and wondering how could you see emptyStateView at the donor side since all the donors in the database have some donation already...

**Comment by kara-o** [comment]
I temporarily changed the logic on Donations.tsx (line 40)/DonationsOrClaims.tsx (line 41) so it would evaluate to false.

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
+		"editor.codeActionsOnSave": {
+			"source.fixAll.eslint": true
+		}
+	},
+	"[typescriptreact]": {
+		"editor.codeActionsOnSave": {
+			"source.fixAll.eslint": true
+		}
+	},
+	"eslint.codeAction.showDocumentation": {
```
Comment (jacksonrya) [inline .vscode/settings.json:28]: I would remove these empty eslint actions though. Make sure if you're customizing your IDE that you do it in your User/ Global settings and have it override the workspace.

File: .vscode/launch.json [hunk 2]
```diff
@@ -1,22 +1,22 @@
 {
-    // Use IntelliSense to learn about possible attributes.
```
Comment (jacksonrya) [inline .vscode/launch.json:2]: Probably revert this file. If you have auto-fix linting on save enabled. You can either save without formatting via the VSCode command `File: Save without formatting`, or just don't stage this file.

**Comment by jacksonrya** [comment]
If the image is visible on mobile, then I think this can be merged in as long as a note is provided within file or an issue is created that explains this. I would consider browser fixes low priority.

File: src/elements/Paragraph/Paragraph.tsx [hunk 3]
```diff
@@ -3,15 +3,19 @@ import { Text } from 'react-native';
 import styles from './Paragraph.styles';
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/elements/Paragraph/Paragraph.tsx:3]: Oh good point!  Will definitely look into that, thanks :)

File: src/screens/DashboardScreen/DashboardScreen.styles.ts [hunk 4]
```diff
@@ -7,9 +7,11 @@ const addButtonSize = 110;
 export default StyleSheet.create({
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/screens/DashboardScreen/DashboardScreen.styles.ts:4]: Done!

File: src/screens/DonorDashboardScreen/DonorDonationScreen/DonorDonationScreen.tsx [hunk 5]
```diff
@@ -20,7 +20,7 @@ import * as colors from '@util/colors';
 import styles from './DonorDonationScreen.styles';
```
[the remark this answers is not in the document]
Comment (kara-o) [reply src/screens/DonorDashboardScreen/DonorDonationScreen/DonorDonationScreen.tsx]: done!

File: .vscode/settings.json (hunk 1 above)
Comment (kara-o) [reply .vscode/settings.json]: Good point!  I reverted this file.

File: .vscode/launch.json [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
-        },
-        {
-            "name": "Debug iOS",
-            "cwd": "${workspaceFolder}",
-            "type": "reactnative",
-            "request": "launch",
-            "platform": "ios"
-        }
-    ]
+	// Use IntelliSense to learn about possible attributes.
```
[the remark this answers is not in the document]
Comment (kara-o) [reply .vscode/launch.json]: Thanks Tao - I reverted both of those .vscode files

File: .vscode/launch.json (hunk 2 above)
Comment (kara-o) [reply .vscode/launch.json]: reverted!

**Comment by kara-o** [comment]
Thanks Jackson and Tao for your reviews!  I addressed the comments and will go ahead and merge.  Will make an issue for the browser image issue.  Thanks!

**Pull request merged**